Shared/Samples/Create and save KML file/CreateAndSaveKMLView.swift

```diff
@@ -0,0 +1,124 @@
+                                mainMenuContent
+                            } else {
+                                // If the geometry editor is started, show the edit menu.
+                                editMenuContent
+                            }
+                        } label: {
+                            Label("Geometry Editor", systemImage: "pencil.tip.crop.circle")
+                        }
+                        
+                        Button {
+                            model.showingFileExporter = true
+                        } label: {
+                            Label("Export File", systemImage: "square.and.arrow.up")
+                        }
+                        .disabled(model.fileExporterButtonIsDisabled)
+                    }
+                }
+            }
+            .fileExporter(isPresented: $model.showingFileExporter, document: model.kmzFile, contentType: .kmz) { result in
+                switch result {
+                case .success:
+                    // We no longer need the file locally.
+                    model.kmzFile.deleteFile()
+                    
+                    // We no longer have a local file so disable file exporter button.
+                    model.fileExporterButtonIsDisabled = true
+                case .failure(let error):
+                    model.error = error
+                }
+            }
+    }
+}
+
+/// A KMZ file that can be used with the native file exporter.
+final class KMZFile: FileDocument {
+    /// The KML document that is used to create the KMZ file.
+    private let document: KMLDocument
+    
+    /// The temporary directory where the KMZ file will be stored.
+    private var temporaryDirectory: URL?
+    
+    /// The temporary URL to the KMZ file.
+    private var temporaryDocumentURL: URL?
+    
+    static var readableContentTypes = [UTType.kmz]
+    
+    /// Creates a KMZ file with a KML document.
+    /// - Parameter document: The KML document that is used when creating the KMZ file.
+    init(document: KMLDocument) {
+        self.document = document
+    }
+    
+    // This initializer loads data that has been saved previously.
+    init(configuration: ReadConfiguration) throws {
+        fatalError("Loading KML files is not supported by this sample")
+    }
+    
+    // This will be called when the system wants to write our data to disk
+    func fileWrapper(configuration: WriteConfiguration) throws -> FileWrapper {
+        guard let temporaryDocumentURL else { return FileWrapper() }
+        return try FileWrapper(url: temporaryDocumentURL)
+    }
+    
+    /// Deletes the temporarily stored KMZ file.
+    func deleteFile() {
+        guard let url = temporaryDirectory else { return }
+        try? FileManager.default.removeItem(at: url)
+    }
+    
+    /// Saves the KML document as a KMZ file to a temporary location.
+    func saveFile() async throws {
+        temporaryDirectory = try? FileManager.default.url(
```

Comment:
Do we need to guard the `temporaryDirectory` here, though it is very unlikely to be `nil`?

Reply:
I see in other samples you used `try!` and the apple doc says this:
<img width="834" alt="Screenshot 2023-06-05 at 4 24 34 PM" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/74069582/04c8cc38-0449-4db0-9e44-3616adc3f4e7">

So I think it is ok to assume there will always be a temporary directory.